reinforcement/algorithm/reinforce.py

```diff
@@ -19,16 +23,21 @@ def sample(self, observation):
 
     def optimize(self, trajectory):
         self._update_rewards_to_return_signals(trajectory)
-        self._policy.fit(trajectory)
+        self._trajectories.append(trajectory)
+        if len(self._trajectories) == self._num_train_trajectories:
+            total = sum(self._trajectories, Trajectory())
```

Comment:
join/chain/concatenate would probably be clearer as python like function instead of sum